packages/next/src/build/webpack/plugins/next-trace-entrypoints-plugin.ts

```diff
@@ -399,29 +495,35 @@ export class TraceEntryPointsPlugin implements webpack.WebpackPluginInstance {
 
             const entryPaths = Array.from(entryModMap.keys())
 
-            const collectDependencies = (mod: any) => {
+            const collectDependencies = async (mod: any) => {
               if (!mod || !mod.dependencies) return
 
               for (const dep of mod.dependencies) {
                 const depMod = getModuleFromDependency(compilation, dep)
 
                 if (depMod?.resource && !depModMap.get(depMod.resource)) {
+                  if (this.flyingShuttle) {
+                    this.traceHashes.set(
+                      depMod.resource,
+                      await getOriginalHash(depMod.resource)
```

Comment:
should we `Promise.all` these dependency hashes?

Reply:
I didn't for this one as it's recursive and don't want to overload the stack